core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java

```diff
@@ -390,8 +390,12 @@ public Response getBucket(@HeaderParam("Authorization") final String authorizati
         } catch (FileDoesNotExistException e) {
           // return the proper error code if the bucket doesn't exist. Previously a 500 error was
           // returned which does not match the S3 response behavior
-          auditContext.setSucceeded(false);
-          throw new S3Exception(e, bucket, S3ErrorCode.NO_SUCH_BUCKET);
+          // - this should never happen since we've called S3RestUtils.checkPathIsAlluxioDirectory()
+          if (prefixParam == null) {
```

Comment:
null or empty string?

Reply:
Hmm good point, let me test that

Reply:
Actually a lot of the inputs to these methods are checking for null strings, I don't know what the guarantees are provided by the service calling this method, so it's not entirely clear what the expected behaviors are

Reply:
Okay well, this code block should never execute anyway because of the guarded `S3RestUtils.checkPathIsAlluxioDirectory(fs, path, auditContext);` check prior to this. I'm just removing this entirely.

Reply:
I guess it could happen if there is a concurrent delete of the folder, but I think this is fine anyway since it will still be linearizable given that S3 doesn't have any atomic removal operations.